app/models/concerns/filterable.rb

```diff
@@ -0,0 +1,13 @@
+module Filterable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
```

Comment:
This is a flexible solution that will send all of the filtering params and then call each scope it the filters are present